mobilizon_bots/publishers/coordinator.py

```diff
@@ -1,104 +1,83 @@
 from dataclasses import dataclass, field
-from enum import IntEnum
-from typing import List
+from uuid import UUID
 
 from mobilizon_bots.event.event import MobilizonEvent
-from mobilizon_bots.publishers import get_active_publishers
-from mobilizon_bots.publishers.abstract import AbstractPublisher
+from mobilizon_bots.models.publication import PublicationStatus
 from mobilizon_bots.publishers.exceptions import PublisherError
 from mobilizon_bots.publishers.telegram import TelegramPublisher
 
 KEY2CLS = {"telegram": TelegramPublisher}
 
 
-class PublisherStatus(IntEnum):
-    WAITING = 1
-    FAILED = 2
-    COMPLETED = 3
-
-
 @dataclass
-class PublisherReport:
-    status: PublisherStatus
+class PublicationReport:
+    status: PublicationStatus
     reason: str
-    publisher: AbstractPublisher
 
 
 @dataclass
 class PublisherCoordinatorReport:
-    reports: List[PublisherReport] = field(default_factory=[])
+    reports: dict[UUID, PublicationReport] = field(default_factory={})
 
     @property
     def successful(self):
-        return all(r.status == PublisherStatus.COMPLETED for r in self.reports)
+        return all(
+            r.status == PublicationStatus.COMPLETED for r in self.reports.values()
+        )
 
     def __iter__(self):
-        return self.reports.__iter__()
+        return self.reports.items().__iter__()
 
 
 class PublisherCoordinator:
-    def __init__(self, event: MobilizonEvent):
-        self.publishers = tuple(KEY2CLS[pn](event) for pn in get_active_publishers())
+    def __init__(self, event: MobilizonEvent, publications: list[tuple[UUID, str]]):
+        self.publications = tuple(
+            (uuid, KEY2CLS[pn](event)) for uuid, pn in publications
+        )
 
     def run(self) -> PublisherCoordinatorReport:
-        invalid_credentials, invalid_event, invalid_msg = self._validate()
-        errors = invalid_credentials + invalid_event + invalid_msg
+        errors = self._validate()
         if errors:
             return PublisherCoordinatorReport(reports=errors)
 
         return self._post()
 
     def _make_successful_report(self):
-        return [
-            PublisherReport(
-                status=PublisherStatus.COMPLETED,
+        return {
+            uuid: PublicationReport(
+                status=PublicationStatus.COMPLETED,
                 reason="",
-                publisher=p,
             )
-            for p in self.publishers
-        ]
+            for uuid, _ in self.publications
```

Comment:
don't call it uuid but call it `publication_id` or whatever it represents

Reply:
now it should be ok